telegram/ext/_jobqueue.py

```diff
@@ -230,11 +254,11 @@ async def callback(context: CallbackContext)
         date_time = self._parse_time_input(when, shift_day=True)
 
         j = self.scheduler.add_job(
-            job.run,
+            self.job_callback,
             name=name,
             trigger="date",
             run_date=date_time,
-            args=(self.application,),
+            args=(self, job),
```

Comment:
Wow :) `args` are passed as args to a callback (self.job_callback), which then passes itself as a callback and `JobQueue` and `Job` as args to this callback. It feels like falling down a tunnel with mirrors ;)

Reply:
That's pretty much the point of this PR :D As the PRs linked in the description show, the current status is that

* what we pass to APS is intransparent to the PTB users, making it hard for them to implement a PTB-conformant JobStore
* Because we use `Job.run` - a bound method - as callback, we have to rely on `function.__self__` in `from_aps_job` but that breaks as soon as the link between the instance of `ext.Job` and `Job.run` get's lost, i.e. if you call `Job.run(some_job)` instead of `some_job.run()`. This makes it event harder to implement APS-logic, especially since `_from_aps_job` is protected so far

The downside of this PR is that we expose the "backend" (APS) more to the user, which makes it a bit harder for us to ditch/upgrade APS if ever needed (APS is preparing v4 since before we incorporated it in ptbv13 :D). Then again, we have a fancy new stability policy in place which makes it rather clear what we'll have to do in that case.

Reply:
Oh, sorry I didn't reply explicitly that I read and acknowledged this :)